### DISCUSSION TITLE
Broadphase/Object Layer Composition

### DISCUSSION THREAD

**Discussion by joshcoyne** [jrouwe/JoltPhysics#650] [Q&A] [answered]
I'm looking for some advice on optimizing the broadphase. Right now I have the following object layers:

OBJECT_STATIC // non moving geo, mostly tri mesh
OBJECT_DYNAMIC // dynamic rigid bodies that collide with static and dynamic geo
OBJECT_DYNAMIC_SINGLE // dynamic rigid body that only collide with OBJECT_STATIC
OBJECT_MOVE_OBSTACLE // kinematic capsules used for character movement avoidance

I also set up matching broadphases for a total of 4 BP's but I'm wondering if this is ill advised. Would it be more efficient instead to have 2 BP's, one static, one dynamic, then rely on object layers in the dynamic set to filter?

**Comment by jrouwe** [comment, marked as answer]
There's no simple answer to that question. It depends on the number of objects in each layer, their distribution and how often they change position. You'll have to profile which one is faster. At the very least you should have 1 large tree with static/infrequent moving objects and 1 smaller tree with dynamic moving objects, but anything beyond that needs to be tested.
